bucket-cache.go

```diff
@@ -170,7 +170,7 @@ func (c Client) getBucketLocationRequest(bucketName string) (*http.Request, erro
 	urlValues.Set("location", "")
 
 	// Set get bucket location always as path style.
-	targetURL := c.endpointURL
+	targetURL := *c.endpointURL
```

Comment:
also needs to be changed in bucket-cache_test.go

Reply:
done